hamilton/graph.py

```diff
@@ -787,8 +788,14 @@ def display(
             deduplicate_inputs,
             display_fields=display_fields,
         )
-        kwargs = {"view": True}
-        if render_kwargs and isinstance(render_kwargs, dict):
+        kwargs = {"view": False, "format": "png"}  # default format = png
+        if output_file_path:  # infer format from path
+            if suffix := pathlib.Path(output_file_path).suffix:
```

Comment:
IMO you should use `is not None` for clarity (though I have a thing against truthiness in python...)

Reply:
The changes in behavior:
1. PNG is the default instead of PDF
2. a non-default format (previously PDF) can be produced without specifying it in `render_kwargs`
3. the format is inferred from "the rightmost `.{string}`

Regarding backwards compatibility and #3, the main foreseeable issue is people who were specifying `dr.display_all_functions("/path/to/file.dot")`, which would generate `file` and `file.dot` that both contain the Graphviz specs.